relay-general/src/user_agent.rs

```diff
@@ -26,18 +26,50 @@ static UA_PARSER: Lazy<UserAgentParser> = Lazy::new(|| {
     )
 });
 
-fn get_user_agent_from_headers(headers: &Headers) -> Option<&str> {
-    for item in headers.iter() {
-        if let Some((ref o_k, ref v)) = item.value() {
-            if let Some(k) = o_k.as_str() {
-                if k.to_lowercase() == "user-agent" {
-                    return v.as_str();
+/// this has both the user agent string and the client hints, useful for the scenarios where
+/// you will use either information from client hints if it exists and if not fall back to
+/// user agent string.
+///
+/// The client hint variable names mirror the name of the headers. '<https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#user_agent_client_hints>'
+#[derive(Default, Debug)]
+pub struct RawUserAgentInfo<'a> {
+    /// the "old style" of a single UA string
```

Comment:
We always start with a capital letter and end with punctuation.

```suggestion
    /// The "old style" of a single UA string.
```